File: check.rb
#!/usr/bin/env ruby

def an(n)
	[Regexp.new("a?"*n + "a"*n), "a"*n]
end

n = ARGV[0].to_i
p, s = an(n)
idxs = p.match(s).offset(0)
puts "n: #{n} match: [#{idxs.join(' ')}]" 

File: check.py
#!/usr/bin/env python2

import re, sys

def an(n): return (re.compile('a?'*n+'a'*n), 'a'*n)

n = int(sys.argv[1])
r, s = an(n)
idxs = r.match(s).span()
print 'n:', n, 'match:', '[%s]' % ' '.join(map(str, idxs))
